File: Java/AlgoExpert/Recursion/Easy/ProductSum.java
import java.util.*;

class Program {
  // Tip: You can use `element instanceof ArrayList` to check whether an item
  // is an array or an integer.
  public static int productSum(List<Object> array) {
    // Write your code here.
    return productSum(array, 1);
  }
	
	public static int productSum(List<Object> array, int multiplier) {
    // Write your code here.
		int sum = 0;
		
		for(int i = 0; i < array.size(); i++){
			if(array.get(i) instanceof ArrayList){
				sum += productSum((List<Object>) array.get(i), multiplier + 1);
			}else if(array.get(i) instanceof Object){
				sum += (int) array.get(i);
			}
		}
		
    return sum * multiplier;
  }
}

File: Python/AlgoExpert/Recursion/Easy/productSum.py
# Tip: You can use the type(element) function to check whether an item
# is a list or an integer.
def productSum(array, depth=1):
    # Write your code here.
    sum = 0
    for element in array:
        if isinstance(element, int):
            sum += element
        else:
            sum += productSum(element, depth + 1)

    return depth * sum
